addons/libkosext2fs/inode.c

```diff
@@ -989,9 +989,13 @@ static ext2_dirent_t *search_indir_23(ext2_fs_t *fs, const uint32_t *iblock,
     uint32_t *buf;
     int i, block_ents;
     ext2_dirent_t *rv;
+    /* Ensure the buffer will hold uint32_ts */
+    size_t bufsize = block_size + (block_size % sizeof(uint32_t)) ? 
+                                  (sizeof(uint32_t) - (block_size % sizeof(uint32_t))) :
+                                  0;
```

Comment:
Using a bitwise operation should be faster than division and a branch.

```suggestion
    /* Round up to ensure the buffer will hold uint32_ts */
    size_t bufsize = (block_size + sizeof(uint32_t) - 1) & ~(sizeof(uint32_t) - 1);
```